docs/backend/http.md

```diff
@@ -0,0 +1,187 @@
+
+- **Domain**: The address of the server (e.g., `www.example.com`).
+
+- **Path**: The path to the specific resource on the server (e.g., `/page/about`).
+
+## Status Codes
+
+HTTP responses are accompanied by status codes that indicate the outcome of the request. Some common status codes include:
+
+- `200 OK`: The request was successful, and the server is returning the requested data.
+
+- `404 Not Found`: The requested resource could not be found on the server.
+
+- `500 Internal Server Error`: An error occurred on the server while processing the request.
+
+## Using HTTP Requests in JavaScript
+
+Now, let's see how you can use JavaScript to make HTTP requests using the Fetch API. We'll provide examples for GET, POST, PUT, and DELETE requests along with their outputs.
+
+### GET Request example:
+
+```javascript
+// GET Request Example
+fetch("https://jsonplaceholder.typicode.com/posts/1")
+  .then((response) => response.json())
+  .then((data) => console.log(data))
+  .catch((error) => console.error("Error:", error));
+```
+
+#### Output:
+
+```javascript
+{
+  "userId": 1,
+  "id": 1,
+  "title": "sunt aut facere repellat provident occaecati excepturi optio reprehenderit",
+  "body": "quia et suscipit\nsuscipit..."
+}
+```
+
+```
+This code sends a GET request to https://jsonplaceholder.typicode.com/posts/1,
+retrieves the data, and logs it to the console.
+```
+
+### POST REQUEST EXAMPLE:
+
+```javascript
+// POST Request Example
+const postData = {
+  userId: 1,
+  id: 101,
+  title: "Sample Post",
+  body: "This is a sample post body.",
+};
+
+fetch("https://jsonplaceholder.typicode.com/posts", {
+  method: "POST",
+  headers: {
+    "Content-Type": "application/json",
+  },
+  body: JSON.stringify(postData),
+})
+  .then((response) => response.json())
+  .then((data) => console.log(data))
+  .catch((error) => console.error("Error:", error));
+```
+
+#### OUTPUT:
+
+```javascript
+{
+  "userId": 1,
+  "id": 101,
+  "title": "Sample Post",
+  "body": "This is a sample post body."
+}
+```
+
+```
+This code sends a POST request to https://jsonplaceholder.typicode.com/posts with JSON data,
+and then logs the response data to the console.
+```
+
+### PUT REQUEST EXAMPLE:
+
+```javascript
+// PUT Request Example
+const updatedData = {
+  id: 1,
+  title: "Updated Post Title",
+  body: "This is the updated post body.",
+};
+
+fetch("https://jsonplaceholder.typicode.com/posts/1", {
+  method: "PUT",
+  headers: {
+    "Content-Type": "application/json",
+  },
+  body: JSON.stringify(updatedData),
+})
+  .then((response) => response.json())
+  .then((data) => console.log(data))
+  .catch((error) => console.error("Error:", error));
+```
+
+#### OUTPUT:
+
+```javascript
+{
+  "userId": 1,
+  "id": 1,
+  "title": "Updated Post Title",
+  "body": "This is the updated post body."
+}
+```
+
+```
+This code sends a PUT request to update an existing resource at https://jsonplaceholder.typicode.com/posts/1.
```

Comment:
we are not doing with our links anymore right?

Reply:
Yeah, the code examples contain many http methods while our endpoint only supports GET, so we can only show one example of our endpoint and the rest relies on `jsonplaceholder`.

Instead of that, I think we should just keep using the `jsonplaceholder` for every example just for clarity. Wdyt?

Reply:
yea yeeaa make sense! They can even try out theirselves with this one so.